Build/downgrade_nsis.ps1

```diff
@@ -2,16 +2,45 @@
 ##  File:  Install-NSIS.ps1
 ##  Desc:  Install NSIS
 ################################################################################
+function Set-MachinePath{
+    [CmdletBinding()]
+    param(
+        [string]$NewPath
+    )
+    Set-ItemProperty -Path 'Registry::HKEY_LOCAL_MACHINE\System\CurrentControlSet\Control\Session Manager\Environment' -Name Path -Value $NewPath
+    return $NewPath
+}
+
+function Add-MachinePathItem
+{
+    [CmdletBinding()]
+    param(
+        [string]$PathItem
+    )
+
+    $currentPath = Get-MachinePath
+    $newPath = $PathItem + ';' + $currentPath
+    return Set-MachinePath -NewPath $newPath
+}
+
+function Get-MachinePath{
+    [CmdletBinding()]
+    param(
+
+    )
+    $currentPath = (Get-ItemProperty -Path 'Registry::HKEY_LOCAL_MACHINE\System\CurrentControlSet\Control\Session Manager\Environment' -Name PATH).Path
+    return $currentPath
+}
 
 $NsisVersion = "3.04"
-Invoke-WebRequest "https://netcologne.dl.sourceforge.net/project/nsis/NSIS%203/${NsisVersion}/nsis-${NsisVersion}-setup.exe" -OutFile "C:\WINDOWS\Temp\nsis-${NsisVersion}-setup.exe"
-Start-Process -Wait -FilePath "C:\WINDOWS\Temp\nsis-${NsisVersion}-setup.exe" -ArgumentList "/S"
+# Get-Help Install-Binary -Full
+
+Install-Binary -Url "https://downloads.sourceforge.net/project/nsis/NSIS%203/${NsisVersion}/nsis-${NsisVersion}-setup.exe" -Type EXE -InstallArgs ('/S')
```

Comment:
using original Install-Binary cmd-let after so much work figuring out what the new params are.